File: Python/class_object_hero.py
class Hero:
    attack_types = {
        'warrior': 'espada',
        'mage': 'magia',
        'monk': 'artes marciais',
        'ninja': 'shuriken'
    }

    def __init__(self, name, age, hero_type):
        self.name = name
        self.age = age
        self.hero_type = hero_type
        self.attack_type = self.attack_types.get(hero_type)

    def attack(self):
        print(f'O {self.hero_type} atacou usando {self.attack_type}.')

hero1 = Hero('Zeno', 135, 'ninja')
hero2 = Hero('Zero', 354, 'mage')

hero1.attack()
hero2.attack()

File: Go/class_object_hero.go
package main

import "fmt"

var attackTypes = map[string]string{
	"warrior": "espada",
	"mage":    "magia",
	"monk":    "artes marciais",
	"ninja":   "shuriken",
}

type Hero struct {
	name       string
	age        int
	heroType   string
	attackType string
}

func NewHero(name string, age int, heroType string) Hero {
	attackType, exists := attackTypes[heroType]
	if !exists {
		attackType = "desconhecido"
	}
	return Hero{
		name:       name,
		age:        age,
		heroType:   heroType,
		attackType: attackType,
	}
}

func (h Hero) Attack() {
	fmt.Printf("O %s atacou usando %s.\n", h.heroType, h.attackType)
}

func main() {
	hero1 := NewHero("Zeno", 432, "ninja")
	hero2 := NewHero("Zero", 622, "mage")

	hero1.Attack()
	hero2.Attack()
}
